Assets/Scripts/Tools/MovementExecutorTool.cs

```diff
@@ -26,15 +28,24 @@ public void OnLeftClick(Transform cameraTransform, bool isRayCastHit, RaycastHit
       _toolPanelController.SetBackgroundColor(GetPanelColor());
     }
 
-    public void OnRightClick(Transform cameraTransform, bool isRayCastHit, RaycastHit raycastHit) =>
+    public void OnRightClick(Transform cameraTransform, bool isRayCastHit, RaycastHit raycastHit)
+    {
       _movementExecutor.ToggleReverse();
+      RefreshToolgunModeText();
+    }
 
     public void OnLeftMouseButtonHeld(Transform cameraTransform) { }
 
     public void OnLeftMouseButtonReleased() { }
 
-    public void OnSelect() => _toolPanelController.SetBackgroundColor(GetPanelColor());
+    public void OnSelect()
+    {
+      _toolPanelController.SetBackgroundColor(GetPanelColor());
+      RefreshToolgunModeText();
+    }
 
     private Color GetPanelColor() => _movementExecutor.MovementEnabled ? Color.green : Color.red;
+
+    private string GetMovementExecutorMode() => _movementExecutor.Reverse() ? "Reverse" : "Forward";
```

Comment:
I would name this GetMovementExecutorModeLabel or such as Mode seems as it should return an actual mode via enum